builder/local_relay.go

```diff
@@ -0,0 +1,263 @@
+package builder
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"html/template"
+	"net/http"
+	"strconv"
+	"sync"
+
+	builderSpec "github.com/attestantio/go-builder-client/spec"
+	consensusspec "github.com/attestantio/go-eth2-client/spec"
+	"github.com/attestantio/go-eth2-client/spec/bellatrix"
+	"github.com/attestantio/go-eth2-client/spec/deneb"
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	eth2UtilBellatrix "github.com/attestantio/go-eth2-client/util/bellatrix"
+	eth2UtilCapella "github.com/attestantio/go-eth2-client/util/capella"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/flashbots/go-boost-utils/bls"
+	"github.com/flashbots/go-boost-utils/utils"
+	"github.com/gorilla/mux"
+)
+
+// TODO (deneb): remove local relay
+
+type ForkData struct {
+	GenesisForkVersion    string
+	BellatrixForkVersion  string
+	GenesisValidatorsRoot string
+}
+
+type FullValidatorData struct {
+	ValidatorData
+	Timestamp uint64
+}
+
+type LocalRelay struct {
+	beaconClient IBeaconClient
+
+	relaySecretKey        *bls.SecretKey
+	relayPublicKey        phase0.BLSPubKey
+	serializedRelayPubkey hexutil.Bytes
+
+	builderSigningDomain  phase0.Domain
+	proposerSigningDomain phase0.Domain
+
+	validatorsLock sync.RWMutex
+	validators     map[PubkeyHex]FullValidatorData
+
+	enableBeaconChecks bool
+
+	bestDataLock   sync.Mutex
+	bestHeader     *deneb.ExecutionPayloadHeader
```

Comment:
`bestHeader` isn't used, can be removed from the local relay also

Reply:
4d085c4f5a5b863199161a7d72a90abdde5d743a
op-builder now supports multiple version. The same has been applied to op-node as well.